pkg/rekor/v2/client.go

```diff
@@ -18,29 +18,107 @@ import (
 	"context"
 	"fmt"
 	"net/url"
+	"slices"
 	"time"
 
-	prototrustroot "github.com/sigstore/protobuf-specs/gen/pb-go/trustroot/v1"
 	tiles_client "github.com/sigstore/rekor-tiles/pkg/client"
 	"github.com/sigstore/rekor-tiles/pkg/client/read"
 	"github.com/sigstore/sigstore-go/pkg/root"
+	"github.com/sigstore/sigstore-go/pkg/tuf"
 	"github.com/sigstore/sigstore/pkg/signature"
 )
 
 type ShardInfo struct {
-	client   *read.Client
-	verifier *signature.Verifier
+	client      *read.Client
+	verifier    *signature.Verifier
+	validityEnd time.Time
 }
 
-func GetRekorShards(ctx context.Context, trustedRoot *root.TrustedRoot, rekorServices []root.Service, userAgent string) (map[string]ShardInfo, string, error) {
-	rekorServices, err := root.SelectServices(rekorServices, root.ServiceConfiguration{Selector: prototrustroot.ServiceSelector_ALL}, []uint32{2}, time.Now())
+func RefreshSigningConfig(tufClient *tuf.Client) (*root.SigningConfig, error) {
+	err := tufClient.Refresh()
+	if err != nil {
+		return nil, fmt.Errorf("error refreshing TUF client: %v", err)
+	}
+
+	signingConfig, err := root.GetSigningConfig(tufClient)
```

Comment:
we should also update the main.go file to use `root.GetSigningConfig(tufClient)` (but ok to do in another PR).